PeakNetwork/Core/DecodeInputOperation.swift

```diff
@@ -0,0 +1,242 @@
+//
+//  DecodingOperation.swift
+//  PeakNetwork
+//
+//  Created by Sam Oakley on 12/02/2019.
+//  Copyright © 2019 3Squared. All rights reserved.
+//
+
+#if os(iOS) || os(tvOS)
+import UIKit
+#else
+import AppKit
+#endif
+import PeakOperation
+import PeakResult
+
+
+/// An operation which takes an Input and decodes it to an Output.
+/// A more specialised version of `MapOperation`, easier to use for this specific task.
+open class DecodeInputOperation<I, O>: ConcurrentOperation, ConsumesResult, ProducesResult {
```

Comment:
Why not just call it `DecodeOperation`? It is implied it will be decoding its input to produce an appropriate output.
```suggestion
open class DecodeOperation<I, O>: ConcurrentOperation, ConsumesResult, ProducesResult {
```

Reply:
👍